File: backend/ria_ml/SmithCase_Chapter11.m
% SmithCase_Chapter11.m
 
% clear all previous variables and close any figures
   clear all;
   close all;
 
% create a new client data structure
   client = client_create( );
% change client data elements as needed  
%    ...
% process the client data structure 
   client = client_process( client );
 
% create a new market data structure
   market = market_create( );
% change market data elements as needed
%   ...
% process the client data structure
   market = market_process( market, client ); 
 
% Create a fixed annuity
   iFixedAnnuity = iFixedAnnuity_create( );
% change fixed annuity data elements as needed
%   ...
% process fixed annuity and update client matrices
   client = iFixedAnnuity_process( iFixedAnnuity, client, market );  
   
% create analysis
   analysis = analysis_create( );
% select desired output
   analysis.plotSurvivalProbabilities = 'y';
% process analysis
   analysis_process( analysis, client, market);

File: backend/ria_ml/SmithCase_Chapter11.py
from client_create import client_create
from client_process import client_process
from market_create import market_create
from market_process import market_process
from iFixedAnnuity_create import iFixedAnnuity_create
from iFixedAnnuity_process import iFixedAnnuity_process
from analysis_create import analysis_create
from analysis_process import analysis_process

client = client_create()
client = client_process(client)

market = market_create()
market = market_process(market, client)

iFixedAnnuity = iFixedAnnuity_create()
client = iFixedAnnuity_process(iFixedAnnuity, client, market)

analysis = analysis_create()
analysis['plotSurvivalProbabilities'] = 'y'
analysis_process(analysis, client, market)
